tests/h/services/group_list_test.py

```diff
@@ -235,50 +108,85 @@ def test_it_returns_empty_list_if_no_user(self, svc):
 
         assert u_groups == []
 
+    @pytest.fixture
+    def user_groups(self, user, factories):
+        return [
+            factories.Group(creator=user, name="Gamma"),
+            factories.Group(creator=user, name="Oomph"),
+            factories.Group(creator=user, name="Beta"),
+            factories.Group(creator=user, name="Alpha"),
+        ]
+
 
 class TestPrivateGroups(object):
-    def test_it_returns_a_users_private_groups(self, svc, user, user_groups):
-        user.groups = user_groups
+    def test_it_retrieves_all_user_groups(self, svc, user):
+        svc.user_groups = mock.Mock(return_value=[])
+
+        svc.private_groups(user)
+
+        svc.user_groups.assert_called_once_with(user)
+
+    def test_it_returns_only_private_groups(self, svc, user, factories):
+        private_group = factories.Group()
+        open_group = factories.OpenGroup()
+        restricted_group = factories.RestrictedGroup()
+
+        user.groups = [private_group, open_group, restricted_group]
 
         p_groups = svc.private_groups(user)
 
-        group_names = [group.name for group in p_groups]
-        assert group_names == ["Alpha", "Beta", "Gamma"]
+        assert private_group in p_groups
+        assert open_group not in p_groups
+        assert restricted_group not in p_groups
```

Comment:
It looks like you're trying to test the contents of a list but ignore the order? In that case the `UnorderedList` matcher may be useful and avoid the possibility of missing an entry by mistake.